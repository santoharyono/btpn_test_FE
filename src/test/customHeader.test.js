import React from 'react';
import renderer from 'react-test-renderer';
import CustomHeader from '../components/customHeader';
import {Appbar} from 'react-native-paper';

describe('<CustomHeader>', () => {
  it('<CustomHeader> render correctly', () => {
    expect(renderer.create(<CustomHeader />)).toMatchSnapshot();
  });

  it('render correct title', () => {
    const title = 'title header';
    const instance = renderer.create(<CustomHeader />);
    const appbarContentInstance = instance.findByType(Appbar.Content);
    expect(appbarContentInstance.props.children).toEqual(title);
  });

  it('render Appbar.BackAction when shouldBack equal true', () => {
    const shouldBack = true;
    const inst = renderer.create(<CustomHeader shouldBack={shouldBack} />);
    const instAppbarBackAction = inst.root.findByType(Appbar.BackAction);
    expect(instAppbarBackAction).toBeDefined();
  });
});
