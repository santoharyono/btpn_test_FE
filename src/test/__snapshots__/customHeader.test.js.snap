// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CustomHeader> <CustomHeader> render correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "backgroundColor": "#6200ee",
        "elevation": 4,
        "zIndex": 0,
      },
      Object {
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 3,
          "width": 0,
        },
        "shadowOpacity": 0.24,
        "shadowRadius": 4,
      },
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#6200ee",
        "elevation": 0,
        "flexDirection": "row",
        "height": 56,
        "marginTop": 0,
        "paddingHorizontal": 4,
      }
    }
  >
    <View
      style={
        Object {
          "width": 48,
        }
      }
    />
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "flex": 1,
            "paddingHorizontal": 12,
          },
          Array [
            false,
            Object {
              "alignItems": "center",
            },
            undefined,
          ],
        ]
      }
    >
      <Text
        accessibilityRole="header"
        accessibilityTraits="header"
        accessible={true}
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
              "textAlign": "left",
            },
            Array [
              Object {
                "color": "#ffffff",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              Object {
                "fontSize": 17,
              },
              undefined,
            ],
          ]
        }
      />
    </View>
    <View
      style={
        Object {
          "width": 48,
        }
      }
    />
  </View>
</RCTSafeAreaView>
`;
